### PR TITLE
Fix regression

### PR DESCRIPTION
**Pull request by Nekotekina**

### CODE REVIEW

**Comment by fifazalata** [comment]
Ni No Kuni still stuck in loading screen after you choose to continue your saved game
[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/1998782/RPCS3.log.gz)

**Comment by theoldsport** [comment]
**Demon's Souls** still hangs on loading screen (after selecting save). I noticed this after #4553 was merged. 
[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/1998793/RPCS3.log.gz)

**Comment by psennermann** [comment]
I confirm that NiNoKuni still doesn't work

E {PPU[0x1000000] Thread (main_thread) [0x01113248]} Stat: 'sys_ppu_thread_join' failed with 0x80010005 : CELL_ESRCH [x4]

**Comment by andredias000** [comment]
Heavy Rain and Beyond: Two Souls still regressed with Memory access violation at 0x0 error

**Comment by Xcedf** [comment]
Ni No Kuni regressed in be5c18cc
under ASMJit hangs after pressing X on this screen
previous commit works fine

**Comment by fifazalata** [comment]
@Xcedf that's the demo not the full game, right?

**Comment by Xcedf** [comment]
yes, but it regressed too

**Comment by Josivan88** [comment]
Fifa Street 3 still slow and crash as cited in https://github.com/RPCS3/rpcs3/issues/4536

**Comment by Rpcs3tester** [comment]
Silent Hill: Downpour now fixed

thanks @Nekotekina

**Comment by Parthorisian** [comment]
>  **Rpcs3tester** commented
Silent Hill: Downpour now fixed

Yes can confirm
**SH: Downpour**[BLUS30565] fixed https://i.imgur.com/BVieZYE.jpg Log [downpourWorkinRPCS3.zip](https://github.com/RPCS3/rpcs3/files/2003198/downpourWorkinRPCS3.zip)

**Comment by Crimsom13** [comment]
regression in earth defense force 2050

**Comment by AniLeo** [comment]
@Crimsom13 @Josivan88 regressions happened in this PR or? I think I've seen them on master

**Comment by Josivan88** [comment]
@AniLeo regression in FIFA Street 3 happened before in 6718, and by the title "Fix regression" regression was not fixed, still present.

File: rpcs3/main.cpp [hunk 1]
```diff
@@ -85,6 +90,14 @@ int main(int argc, char** argv)
 	qputenv("QT_AUTO_SCREEN_SCALE_FACTOR", "1");
 #endif
 
+#ifdef __linux__
+	struct ::rlimit rlim;
+	rlim.rlim_cur = 4096;
+	rlim.rlim_max = 4096;
+	if (::setrlimit(RLIMIT_NOFILE, &rlim) != 0)
```
Comment (jbeich) [inline rpcs3/main.cpp]: Is this still necessary after 1193ab0b89da?

Comment (Nekotekina) [reply rpcs3/main.cpp]: Yes.

**Pull request merged**